js/parser.js

```diff
@@ -231,7 +231,11 @@ var ERMrest = (function(module) {
             // columns in the filters are on the projection table
             // use the mapped columns to replace column name to the linked table's column names
             if (colMapping) {
-                _replaceFilterColumns(this._filter, colMapping);
+                var res = _replaceFilterColumns(this._filter, colMapping);
+                if (res === -1) {
+                    console.log("WARNING: Unable to convert filter.");
```

Comment:
this error message would be more informative if you would append to it the offending filter.